app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt

```diff
@@ -58,4 +109,86 @@ class LoginFragment : Fragment() {
         activity?.finish()
     }
 
+    private fun saveCredential(credential: Credential) {
+        isResolving = false
+        Auth.CredentialsApi.save(googleApiClient, credential).setResultCallback { status ->
+
+            if (status.isSuccess) {
+                Timber.d("Credential saved")
+                redirectToMain()
+            } else {
+                Timber.d("Attempt to save credential failed ${status.statusMessage} ${status.statusCode}")
+                resolveResult(status, SAVE_DATA)
+            }
+        }
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+
+        if (requestCode == FETCH_DATA) {
+            mode = FETCH_DATA
+            val credential = data?.getParcelableExtra<Credential>(Credential.EXTRA_KEY)
+            credential?.let { processRetrievedCredential(it) }
+        } else if (requestCode == SAVE_DATA) {
+            mode = SAVE_DATA
+            redirectToMain()
+        }
+    }
+
+    private fun resolveResult(status: Status, requestCode: Int) {
+        if (isResolving) {
+            return
+        }
+
+        if (status.hasResolution()) {
+            try {
+                startIntentSenderForResult(status.resolution.intentSender, requestCode, null, 0, 0, 0, null)
+                isResolving = true
+            } catch (e: IntentSender.SendIntentException) {
+                Timber.e(e)
+            }
+        } else {
+            Timber.e("Resolution Failed!")
+        }
+    }
+
+    private fun requestCredentials() {
+        isRequesting = true
+
+        val request = CredentialRequest.Builder()
+                .setPasswordLoginSupported(true)
+                .build()
+
+        Auth.CredentialsApi.request(googleApiClient, request).setResultCallback { credentialRequestResult ->
+            isRequesting = false
+            val status = credentialRequestResult.status
+            if (status.isSuccess) {
+                val credential = credentialRequestResult.credential
+                processRetrievedCredential(credential)
+            } else if (status.statusCode == CommonStatusCodes.RESOLUTION_REQUIRED) {
+                resolveResult(status, FETCH_DATA)
+            }
+        }
+    }
+
+    private fun processRetrievedCredential(credential: Credential) {
+        loginActivityViewModel.login(credential.id.nullToEmpty(), credential.password.nullToEmpty())
+        username.setText(credential.id)
+        password.setText(credential.password)
+    }
+
+    override fun onPause() {
+        activity?.let { googleApiClient?.stopAutoManage(it) }
+        googleApiClient?.disconnect()
+        super.onPause()
+    }
+
+    override fun onResume() {
+        if (googleApiClient?.isConnected == false) {
+            googleApiClient?.connect()
+        }
+        super.onResume()
+    }
```

Comment:
OK, this fragment is doing a lot of things, wrap the Google Auth stuff in a separate class

Reply:
There seems to be a problem after moving all functions to the class. The onActivityResult never gets fired. It all works fine, when we open the page, we are shown a list of accounts, and when we enter a new credential, it gets saved. But, the automatic login is not occuring as it is handled in onActivityResult. Can you help me on this?